resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -164,6 +165,9 @@ private[spark] class Client(
         }
       } finally {
         kubernetesResourceCleaner.deleteAllRegisteredResourcesFromKubernetes(kubernetesClient)
+        if (loggingWatch.nonEmpty) {
+          loggingWatch.get.shutdown()
```

Comment:
@mccheah SGTM. Let me close this PR. We can always revisit when and if there are future issues.